### DISCUSSION TITLE
Unable to change Cockpit listening port

### DISCUSSION THREAD

**Discussion by itm1960** [cockpit-project/cockpit#18475] [Q&A] [answered]
I'm trying to change the listening port to 9091. Following instructions elsewhere, I've put this content into /etc/systemd/system/cockpit.socket.d/listen.conf:

```
[Socket]
ListenStream=
ListenStream=443
```
I then did:
```
$ sudo systemctl daemon-reload
$ sudo systemctl restart cockpit.socket
```
...but Cockpit is still listening on port 9090. Any idea what I might have done wrong?

**Comment by martinpitt** [comment]
Can you please check the output of `systemctl status cockpit.socket` and `systemctl cat cockpit.socket` to ensure it took your drop-in into account? Do you perhaps still have a running cockpit session, so that `cockpit.service` is still active? Try to stop that, if it's still running.

**Reply by sjagtap** [reply]
Hi @martinpitt , I want to change the listening port for cockpit to 443.
I followed the steps mentioned in this thread. I am unable to access cockpit using "https://server-ip:443". However when I try accessing cockpit using "https://localhost:443", it runs successfully.
What more needs to be done to be able to access cockpit on the server from any browser from anywhere using its management ip?
Eg: "https://mgmt-ip:443"

**Comment by itm1960** [comment]
I stopped and restarted the cockpit socket (using systemctl disable/enable --now cockpit.socket), then ran those commands. The output is below. At the moment cockpit now not accessible on either 9090 or 9091 (and both of those ports have been allowed through the firewall using "sudo ufw allow").
```
$ systemctl status cockpit.socket
● cockpit.socket - Cockpit Web Service Socket
     Loaded: loaded (/lib/systemd/system/cockpit.socket; enabled; vendor preset: enabled)
    Drop-In: /etc/systemd/system/cockpit.socket.d
             └─listen.conf
     Active: active (running) since Thu 2023-03-09 10:47:55 GMT; 23s ago
   Triggers: ● cockpit.service
       Docs: man:cockpit-ws(8)
     Listen: [::]:9090 (Stream)
    Process: 14289 ExecStartPost=/usr/share/cockpit/motd/update-motd  localhost (code=exited, status=0/SUCCESS)
    Process: 14298 ExecStartPost=/bin/ln -snf active.motd /run/cockpit/motd (code=exited, status=0/SUCCESS)
      Tasks: 0 (limit: 2264)
     Memory: 1.0M
     CGroup: /system.slice/cockpit.socket

Mar 09 10:47:55 HP-MICROSERVER systemd[1]: Starting Cockpit Web Service Socket.
Mar 09 10:47:55 HP-MICROSERVER systemd[1]: Listening on Cockpit Web Service Socket.
```
```
$ systemctl cat cockpit.socket
# /lib/systemd/system/cockpit.socket
[Unit]
Description=Cockpit Web Service Socket
Documentation=man:cockpit-ws(8)
Wants=cockpit-motd.service

[Socket]
ListenStream=9090
ExecStartPost=-/usr/share/cockpit/motd/update-motd '' localhost
ExecStartPost=-/bin/ln -snf active.motd /run/cockpit/motd
ExecStopPost=-/bin/ln -snf /usr/share/cockpit/motd/inactive.motd /run/cockpit/motd

[Install]
[WantedBy=sockets.target](http://wantedby%3Dsockets.target/)

# /etc/systemd/system/cockpit.socket.d/listen.conf
[Socket]
Listenstream=
Listenstream=9091
```

**Comment by martinpitt** [comment, marked as answer]
>  Listenstream=

A-ha! Please try to capitalize it correctly, with `ListenStream=...` in /etc/systemd/system/cockpit.socket.d/listen.conf

**Reply by itm1960** [reply]
Aaaaaah.....that was it (doh!). Many thanks!
